spec/mongo/query_cache_spec.rb

```diff
@@ -0,0 +1,341 @@
+require 'spec_helper'
+
+describe Mongo::QueryCache do
+
+  around do |spec|
+    Mongo::QueryCache.clear_cache
+    Mongo::QueryCache.cache { spec.run }
+  end
+
+  before do
+    authorized_collection.delete_many
+  end
+
+  let(:subscriber) { EventSubscriber.new }
+
+  let(:client) do
+    authorized_client.tap do |client|
+      client.subscribe(Mongo::Monitoring::COMMAND, subscriber)
+    end
+  end
+
+  let(:authorized_collection) { client['collection_spec'] }
+
+  describe '#enabled' do
+
+    context 'when query cache is disabled' do
+
+      before do
+        Mongo::QueryCache.enabled = false
+      end
+
+      it 'disables the query cache' do
+        expect(Mongo::QueryCache.enabled?).to be(false)
+      end
+    end
+
+    context 'when query cache is enabled' do
+
+      before do
+        Mongo::QueryCache.enabled = true
+      end
+
+      it 'enables the query cache' do
+        expect(Mongo::QueryCache.enabled?).to be(true)
+      end
+    end
+  end
+
+  describe '#cache' do
+
+    before do
+      Mongo::QueryCache.enabled = false
+      authorized_collection.insert_one({ name: 'testing' })
+      authorized_collection.find(name: 'testing').to_a
+    end
+
+    let(:events) do
+      subscriber.command_started_events('find')
+    end
+
+    it 'executes the block with the query cache enabled' do
```

Comment:
These tests are looking good! I think this might be a good time to talk about integration vs. unit tests. 

Unit tests generally test just one method (aka a unit lol) to check that you get the expected output if you give a certain input. Integration tests test larger features or groups of components for behavior.

For the `#cache` method, a solid unit test might be something like:

```
it 'turns the query cache on inside the block' do
  Mongo::QueryCache.cache do
    expect(Mongo::QueryCache.enabled?).to be true
  end

  expect(Mongo::QueryCache.enabled?).to be false
end
```

I think the tests you have for `#cache` and `#uncached` (as well as the querying tests in the rest of this file) are really good integration tests, and I was wondering what you'd think about moving them over to a file at `spec/integration/query_cache_spec.rb`?